test/module-test.cc

```diff
@@ -63,3 +73,507 @@ TEST(module_test, macros) {
 #endif
   EXPECT_FALSE(macro_leaked);
 }
+
+// The following is less about functional testing (that's done elsewhere)
+// but rather visibility of all client-facing overloads, reachability of
+// non-exported entities, name lookup and overload resolution within
+// template instantitions.
+// Excercise all exported entities of the API at least once.
+// Instantiate as many code paths as possible.
+
+TEST(module_test, to_string) {
+  EXPECT_EQ("42", fmt::to_string(42));
+  EXPECT_EQ("42", fmt::to_string(42u));
+  EXPECT_EQ("42", fmt::to_string(short{42}));
+  EXPECT_EQ("42", fmt::to_string(unsigned short{42u}));
+  EXPECT_EQ("42", fmt::to_string(42L));
+  EXPECT_EQ("42", fmt::to_string(42uL));
+  EXPECT_EQ("42", fmt::to_string(42LL));
+  EXPECT_EQ("42", fmt::to_string(42uLL));
+  EXPECT_EQ("42", fmt::to_string(42.0));
+  EXPECT_EQ("42", fmt::to_string(42.0f));
+  EXPECT_EQ("42", fmt::to_string(42.0L));
+  EXPECT_EQ("42", fmt::to_string("42"));
+  EXPECT_EQ("a", fmt::to_string('a'));
+  EXPECT_EQ("97", fmt::to_string(unsigned char{'a'}));
+  EXPECT_EQ("97", fmt::to_string(signed char{'a'}));
+  EXPECT_EQ("true", fmt::to_string(true));
+  EXPECT_EQ("0x2a", fmt::to_string((void *)42));
+}
+
+TEST(module_test, format) {
+  EXPECT_EQ("42", fmt::format("{}", 42));
+  EXPECT_EQ("42", fmt::format("{:}", 42u));
+  EXPECT_EQ("+42", fmt::format("{:+d}", short{42}));
+  EXPECT_EQ("42", fmt::format("{:d}", unsigned short{42u}));
+  EXPECT_EQ("-42", fmt::format("{0}", -42L));
+  EXPECT_EQ("2a", fmt::format("{0:x}", 42uL));
+  EXPECT_EQ("***42", fmt::format("{:*>5}", 42LL));
+  EXPECT_EQ("0X2A", fmt::format("{:#X}", 42uLL));
+  EXPECT_EQ("===42.000000===", fmt::format("{:=^15f}", 42.0));
+  EXPECT_EQ("00000042.1", fmt::format("{:<010.5G}", 42.1f));
+  EXPECT_EQ(" +4.21235e+11", fmt::format("{:+{}.{}e}", 42.1234567e10L, 13, 5));
+  EXPECT_EQ("-0x1.5000000000000p+5", fmt::format("{: a}", -42.0));
+  constexpr auto nan = std::numeric_limits<double>::quiet_NaN();
+  EXPECT_EQ("-nan-", fmt::format("{:-^5}", nan));
+  constexpr auto inf = std::numeric_limits<double>::infinity();
+  EXPECT_EQ("-INF-", fmt::format("{:-^5F}", inf));
+  EXPECT_EQ("-INF ", fmt::format("{: 5G}", -inf));
+  EXPECT_EQ("42", fmt::format("{:s}", "42"));
+  EXPECT_EQ("a", fmt::format("{:c}", 'a'));
+  EXPECT_EQ("a", fmt::format("{:c}", unsigned char{'a'}));
+  EXPECT_EQ("a", fmt::format("{:c}", signed char{'a'}));
+  EXPECT_EQ("false", fmt::format("{}", false));
+  EXPECT_EQ("0B1", fmt::format("{:#B}", true));
+  EXPECT_EQ("0x2a", fmt::format("{:p}", (void *)42));
+  EXPECT_EQ("1234", fmt::format("{:L}", 1234));
```

Comment:
For example here we don't need to test all argument types. Since they all go through the same code path, it is very unlikely that one of them works and another doesn't. I suggest picking one type (e.g. int) and testing public methods with it. In the unlikely case that we find some problem in the future, we can add a regression test.